src/graphnet/utils/cluster/condor.py

```diff
@@ -0,0 +1,843 @@
+                if metricsNode:
+
+                    # Read individual details from the metrics file
+                    self.numJobs = metricsNode["jobs"]
+                    self.numJobsSucceeded = metricsNode["jobs_succeeded"]
+                    self.numJobsFailed = metricsNode["jobs_failed"]
+                    self.start_time = metricsNode["start_time"]
+                    self.end_time = metricsNode["end_time"]
+                    self.duration = metricsNode["duration"]
+                    # self.jobsCombinedDuration = metricsNode['total_job_time'] #TODO replace once DAGman actually writes this variable, currently always 0
+                    self.dag_status = metricsNode["dag_status"]
+
+                else:
+                    raise Exception(
+                        'DAGMan metrics file "%s" cannot be parsed as a JSON file'
+                        % metricsJsonFilePath
+                    )
+
+        else:
+            raise Exception(
+                'DAGMan metrics file "%s" does not exist' % metricsJsonFilePath
+            )
+
+
+# Subclass containing .out file information
+class DAGManMetricsOutputFileInfo(object):
+    def __init__(self, dagmanOutputFilePath):
+        self.returnCode = None
+        self.parse(dagmanOutputFilePath)
+
+    def parse(self, dagmanOutputFilePath):  # TODO parse more of this file
+
+        # Check submit exists, open it, and read it into a string
+        if os.path.exists(dagmanOutputFilePath):
+
+            # Parse the return status code
+            self.returnCode = self.getReturnCode(dagmanOutputFilePath)
+
+        else:
+            raise Exception(
+                'DAGMan .out file "%s" does not exist' % dagmanOutputFilePath
+            )
+
+    def fullyParsed(self):
+        return self.returnCode != None
+
+    def getReturnCode(self, dagmanOutputFilePath):
+
+        # Get the last line of the file (which has the status, assuming it has completed)
+        statusLines = tail(dagmanOutputFilePath, 1)
+
+        # Check got the one line I expected
+        if len(statusLines) == 1:
+
+            statusLine = statusLines[0]
+
+            # Parse status from this line
+            statusKey = "EXITING WITH STATUS"
+            if statusKey in statusLine:
+
+                # Extract the status key and return it
+                statusKeyStartIndex = statusLine.find(statusKey)
+                statusCodeStartIndex = (
+                    statusKeyStartIndex + len(statusKey) + 1
+                )  # +1 for the space after it
+                if len(statusLine) > (statusCodeStartIndex + 1):
+                    return int(
+                        statusLine[
+                            statusCodeStartIndex : statusCodeStartIndex + 1
+                        ]
+                    )
+
+        # If here, failed in parsing return code (maybe DAG is still running, or died in an uncontrolled way)
+        return None
```

Comment:
I don't see this class referenced elsewhere (apart from one or more places that are not themselves references elsewhere)?
```suggestion
```